docs/02-guides/customize-wysiwyg-platform.mdx

```diff
@@ -0,0 +1,200 @@
+information in your widget. For instance, if the widget is
+`{% raw %}{{ widget type="acme/product-preview" sku="VSK12" }}{% endraw %}` you
+will want to fetch the product associated with the given SKU.
+
+#### Register your widget type at the GraphQL level
+
+1.  First make sure that your dependencies are up to date in your GraphQL module
+    ```ts title="extensions/acme-extension/modules/wysiwyg/index.ts"
+    export default createGraphQLModule({
+    namespace: "Magento2/Cms",
+    // highlight-start
+    dependencies: [
+      "Magento2/Wysiwyg", // ensure that Widget related features are available
+      "Magento2/Catalog/Product" // ensure that you can fetch a product in your Wysiwyg data
+    ],
+    // highlight-end
+    ```
+1.  Then you need to register a new WidgetData type in your schema
+    ```ts title="extensions/acme-extension/modules/wysiwyg/index.ts"
+    typeDefs: /* GraphQL */ `
+      type WidgetProductPreviewData implements WidgetData {
+        dataId: ID
+        product: Product
+      }
+    `,
+    ```
+1.  After that you can setup the resolvers to tell GraphQL how to fetch the
+    `product` field
+    ```ts title="extensions/acme-extension/modules/wysiwyg/resolvers.ts"
+    export default {
+      WidgetProductPreviewData: {
+        product: ({ node }, _, { loaders }) => {
+          const productSkuAttribute = node.attrs.find(
+            ({ name }) => name === "sku"
+          );
+          if (!productSkuAttribute) {
+            return null;
+          }
+          return loaders.Product.load(productSkuAttribute.value);
+        },
+      },
+    };
+    ```
+1.  Finally register the widget type from Magento and associate it with the
+    GraphQL type
+
+    ```ts title="extensions/acme-extension/modules/wysiwyg/runtime.ts"
+      // ...
+      contextEnhancer: ({ loaders }) => {
+        loaders.MagentoWidget.registerWidgetType(
+          // the tag name in your HTML
+          "acme/product-preview",
+          // The associated GraphQL type name
+          "WidgetProductPreviewData"
+        );
+      },
+      // ...
+    ```
+
+#### Get the data in your component
+
+1.  Override the
+    `theme/modules/Wysiwyg/MagentoWysiwyg/MagentoWysiwygFragment.gql` to fetch
+    the new data needed for your widget
+    ```graphql title="theme/modules/Wysiwyg/MagentoWysiwyg/MagentoWysiwygFragment.gql"
+    fragment MagentoWysiwygFragment on MagentoWysiwyg {
+      childNodes
+      data {
+        dataId
+        ... on WysiwygWidgetData {
+          data {
+            ... on WidgetInvalidData {
+              dataId
+            }
+            // highlight-start
+            ... on WidgetProductPreviewData {
+              product {
+                sku
+                name
+              }
+            }
+            // highlight-end
+          }
+        }
+      }
+    }
+    ```
+1.  Use the fetched data in the `data` props in your final widget component (the
+    `./path/to/ProductPreview.js` mentioned above)
```

Comment:
I can't find where the ProductPreview is mentioned?

It's not clear; 
- there is a `./path/to/ProductPreview.js` where is the `/path/to`? 
- is this a component, a function, a config?

Reply:
It was mentioned in the component lazyloading part. Now it's in the `lazy(() => import("theme/path/to/ProductPreview.js"))`